activiti-cloud-service-common/activiti-cloud-service-messaging-config/src/main/java/org/activiti/cloud/common/messaging/config/ConnectorConfiguration.java

```diff
@@ -0,0 +1,137 @@
+
+                            GenericHandler<Message> handler = (message, headers) -> {
+                                FunctionInvocationWrapper function = functionFromDefinition(beanName);
+                                Object result = function.apply(message);
+
+                                Message<?> response = null;
+                                if (result != null) {
+                                    response = MessageBuilder.withPayload(result)
+                                        .build();
+                                    String destination = headers.get(responseDestination.get(), String.class);
+
+                                    if (StringUtils.hasText(destination)) {
+                                        getStreamBridge().send(destination, response);
+                                        return null;
+                                    }
+                                }
+
+                                return response;
+                            };
+
+                            GenericSelector<Message<?>> selector = Optional.ofNullable(beanFactory.findAnnotationOnBean(beanName, ConnectorBinding.class))
+                                .map(ConnectorBinding::condition)
+                                .filter(StringUtils::hasText)
+                                .map(resolveExpression)
+                                .map(ExpressionEvaluatingSelector::new)
+                                .orElseGet(() -> new ExpressionEvaluatingSelector("true"));
+
+                            IntegrationFlow connectorFlow = IntegrationFlows.from(getGatewayInterface(hasOutput),
+                                    (gateway) -> gateway.beanName(functionName)
+                                        .replyTimeout(0L))
+                                .log(LoggingHandler.Level.INFO, functionName + ".integrationRequest")
+                                .filter(selector)
+                                .handle(Message.class, handler)
+                                .log(LoggingHandler.Level.INFO, functionName + ".integrationResult")
+                                .bridge()
+                                .get();
+
+                            String inputChannel = streamFunctionProperties.getInputBindings(functionName)
+                                .stream()
+                                .findFirst()
+                                .orElse(functionName);
+
+                            IntegrationFlow inputChannelFlow = IntegrationFlows.from(inputChannel)
+                                .gateway(connectorFlow, spec -> spec.replyTimeout(0L).errorChannel("errorChannel"))
+                                .get();
+
+                            integrationFlowContext.registration(inputChannelFlow)
+                                .register();
+                        });
+                }
+                return bean;
+            }
```

Comment:
There is a lot of similarity between this block and the equivalent in [ConditionalFunctionBindingConfiguraiton](https://github.com/Activiti/activiti-cloud/pull/910/files#diff-b2eee1f146d40756788ee2bc81bb1761c6b22460370c51da104425b318d6eeb9R62-R109). I was wondering if part of the logic could not be extracted and reused in a future iteration.

Reply:
Thank you for pointing it out @erdemedeiros. we are going to refactor in a future iteration.